docs/source/user_guide/transform_apply.rst

```diff
@@ -100,6 +100,9 @@ treated that it belongs to a new different DataFrame. See also
   :align: center
   :width: 650
 
+In case of :func:`DataFrame.koalas.apply_batch`, when the given function has the return type annotated, the original index of the DataFrame will be lost and then a default index will be attached to the result DataFrame. Please be careful about configuring default index. See also
```

Comment:
I guess `DataFrame.apply` as well? Could you confirm?

Reply:
I guess we should rather put this in the each function's doc?

- https://koalas.readthedocs.io/en/latest/reference/api/databricks.koalas.DataFrame.apply.html
- https://koalas.readthedocs.io/en/latest/reference/api/databricks.koalas.DataFrame.koalas.apply_batch.html

Reply:
What about `DataFrame.koalas.transform_batch` with the UDF returning `DataFrame`?

Reply:
Updated!

I added GroupBy.transform as well.

Reply:
ah, then GroupBy.apply too?

Reply:
Ah good catch! I was greping 'retain_index=False` so missed this one. Updated!